compiler/rustc_middle/src/query/plumbing.rs

```diff
@@ -24,8 +23,8 @@ pub struct DynamicQuery<'tcx, C: QueryCache> {
     pub eval_always: bool,
     pub dep_kind: DepKind,
     pub handle_cycle_error: HandleCycleError,
-    pub query_state: FieldOffset<QueryStates<'tcx>, QueryState<C::Key>>,
-    pub query_cache: FieldOffset<QueryCaches<'tcx>, C>,
+    pub query_state: usize,
+    pub query_cache: usize,
```

Comment:
Please add comments explaining the meaning of these fields. Currently that seems quite tricky to figure out.

Reply:
Added the comments